WordPress/Classes/ViewRelated/Stats/Traffic/StatsTrafficDatePickerViewModel.swift

```diff
@@ -41,7 +41,9 @@ class StatsTrafficDatePickerViewModel: ObservableObject {
     func formattedCurrentPeriod() -> String {
         let dateFormatter = period.dateFormatter
         if period == .week, let week = StatsPeriodHelper().weekIncludingDate(date) {
-            return "\(dateFormatter.string(from: week.weekStart)) - \(dateFormatter.string(from: week.weekEnd))"
+            let weekEndFormatter = DateFormatter()
+            weekEndFormatter.setLocalizedDateFormatFromTemplate("MMM d, yyyy")
+            return "\(dateFormatter.string(from: week.weekStart)) - \(weekEndFormatter.string(from: week.weekEnd))"
```

Comment:
```suggestion
            let yearFormatter = DateFormatter()
            yearFormatter.setLocalizedDateFormatFromTemplate("yyyy")
            return "\(dateFormatter.string(from: week.weekStart)) - \(weekEndFormatter.string(from: week.weekEnd)), \(yearFormatter.string(from: week.weekEnd))"
```

What about keeping the old return but adding year end of it? This way, we wouldn't copy `MMM d` to `weekEndFormatter`.

Reply:
I wasn't sure if the `,` should be hard-coded. Both in your suggestion and in the current date format string, there's a hard-coded `,` so what I found isn't specific to your suggestion but seems relevant.


| Locale | Explicit `,` in format | An alternative: use system-defined `dateFormatter.dateStyle = .medium` format |
| - | - | - |
| Turkish (tr_TR) | Mar 6, 2024 | 6 Mar 2024 |
| Spanish (es_PY) | mar. 6, 2024 | 6 mar. 2024 |

In both cases, the `,` shouldn't be used (please correct me if I'm wrong about `tr_TR`. To improve localization, I would like to try using the system-defined format in a separate issue which I'll open to list some accessibility and localization improvements. Using the system-defined format would also remove the duplicate definitions of the format, which is what you raised here and is a very valid point.

Reply:
Here's the separate issue: https://github.com/wordpress-mobile/jetpack-issue-repo/issues/22

Reply:
Good point. I'm not sure if `Mar 6, 2024` is valid for Turkish, but I agree that we should use system-defined formats. Thanks for opening the issue.